lib/src/sql/index.rs

```diff
@@ -20,37 +23,91 @@ pub enum Index {
 	/// Unique index
 	Uniq,
 	/// Index with Full-Text search capabilities
-	Search {
-		az: Ident,
-		hl: bool,
-		sc: Scoring,
-		order: u32,
-	},
+	Search(SearchParams),
+	/// M-Tree index for distance based metrics
+	MTree(MTreeParams),
```

Comment:
@rushmorem I have updated this structure using a tuple variant embedding a struct, rather than using a struct variant. For the serialization implementation, what would be the best practice?
The current (failing) implementation is here:
https://github.com/surrealdb/surrealdb/blob/02a55cafecf572d349496956212243c92fc6fa1b/lib/src/sql/value/serde/ser/index/mod.rs